src/screens/CriteriaScreen.tsx

```diff
@@ -0,0 +1,157 @@
+      value: false,
+      requiredValue: false,
+    },
+  ]
+
+  let [consentCriteria, setConsentCriteria] = useState<ExperimentCriteria>(
+    consentCriteriaExample,
+  )
+
+  // Utility function to update the criteria object
+  const updateCriteria = (name: string, value: boolean) => {
+    setConsentCriteria(
+      consentCriteria.map((criteria) => {
+        if (criteria.name == name) {
+          return {
+            ...criteria,
+            value,
+          }
+        }
+
+        return criteria
+      }),
+    )
+  }
+
+  // Utility function to check the validity of consent data
+  const onContinue = () => {
+    // Check if any of the answers make participant incompatible
+    const invalidCriterion = consentCriteria.find(
+      (criterion) =>
+        criterion.requiredValue !== undefined &&
+        criterion.value != criterion.requiredValue,
+    )
+    // Proceed or redirect
+    if (invalidCriterion) {
+      // TODO: Switch to redirect to ScreenOut Screen
+      Alert.alert(
+        'Thank you!',
+        'Unfortunatly you are not an ideal participant for our study, Thank you for time!',
+      )
+      onFailCriteria?.()
+    } else {
+      onPassCritera?.()
+    }
+  }
+
+  return (
+    <>
+      <ScrollView
+        scrollEventThrottle={16}
+        contentInsetAdjustmentBehavior="automatic"
+        style={{
+          position: 'absolute',
+          height: '100%',
+        }}
+      >
+        <Box flex={1} paddingHorizontal={6} paddingTop={10} paddingBottom={10}>
+          {/* Experiment Description */}
+          <Markdown marginBottom={4}>{experimentDescription}</Markdown>
+
+          {/* Loop over each consent criteria */}
+          {consentCriteria.map((criterion) => (
+            <Box key={criterion.name} paddingTop={2} paddingBottom={8}>
+              <Markdown>{criterion.description}</Markdown>
+              <CriterionToggle
+                name={criterion.name}
+                value={criterion.value}
+                onChange={updateCriteria}
+                marginTop={4}
+              />
+            </Box>
+          ))}
+
+          <Box borderTopColor="lightGrey" borderTopWidth={2} paddingTop={6}>
+            {/* Small feature text to remined them to check answers */}
+            <Markdown>{postCriteriaText}</Markdown>
+            <Button
+              testID="ContinueButton"
+              label="Continue"
+              backgroundColor="purple"
+              paddingVertical={4}
+              borderRadius="m"
+              marginTop={4}
+              activeOpacity={0.6}
+              onPress={() => onContinue()}
+            />
+            <Button
+              testID="ExitButton"
+              label="Exit Experiment"
+              marginTop={2}
+              textProps={{ color: 'white' }}
+              onPress={() => onExit?.()}
+              borderRadius="m"
+              backgroundColor="red"
+              paddingVertical={4}
```

Comment:
Not sure if this fits in this PR but it would be good to use variants for buttons so you wouldn't have to keep specifying the same props over and over again.

Reply:
Yeah, No is a good time to clean up